### ISSUE TITLE
openSUSE build

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#2393]
.Net Core now has support for openSUSE, we should have builds of PowerShell Core available

**Comment by AngelCalvo** [comment]
I like to see a timeline...thx

**Comment by andschwa** [comment]
For anyone who wants to get started on this, the minimum requirements would include:

To get it working:
- [ ] Add OpenSUSE Runtime Identifiers to all relevant `project.json` `runtimes` sections
- [ ] Ensure all [Pester tests](https://github.com/PowerShell/PowerShell/tree/master/test/powershell) pass on a base install of OpenSUSE

To get it published:
- [ ] Update [`New-UnixPackage`](https://github.com/PowerShell/PowerShell/blob/f62dd4bf54020d3da05f8e6825d624ceaa2a69b6/build.psm1#L1173) to generate an OpenSUSE package
- [ ] Update [`download.sh`](https://github.com/PowerShell/PowerShell/blob/master/tools/download.sh) to install the package (there's a race condition here because we download packages from GitHub... but until it's made and published that can't be done)
- [ ] Create a pair of [Dockerfiles](https://github.com/PowerShell/PowerShell/tree/master/docker) to build PowerShell on an [OpenSUSE container](https://hub.docker.com/r/library/opensuse/tags/) (same race condition)
- [ ] Update [Linux installation docs](https://github.com/PowerShell/PowerShell/blob/master/docs/installation/linux.md)

The PowerShell maintainers are adding this to GitHub and Digital Ocean's [Hacktoberfest](https://hacktoberfest.digitalocean.com/) for anyone wanting to participate.

**Comment by HemantMahawar** [comment]
@PowerShell/powershell-committee recommendation is to support both, with priority to SLES (13.2).

**Comment by rainerklier** [comment]
please also support opensuse tumbleweed.
there already is a dotnet core repository which features dotnet core 1.1.0:
http://download.opensuse.org/repositories/home:/linnaea/openSUSE_Factory

**Comment by iSazonov** [comment]
@SteveL-MSFT We still want support for OpenSUSE 42.1 so maybe label `Up-for-Gabs`?

**Comment by SteveL-MSFT** [comment]
@iSazonov it's not Up-for-Grabs since it's assigned to @TravisEz13

**Comment by TravisEz13** [comment]
# Checklist for OpenSUSE 42.1

To get it working:
- [ ] Add OpenSUSE [Runtime Identifiers](https://docs.microsoft.com/en-us/dotnet/articles/core/rid-catalog) to all relevant `project.json` `runtimes` sections
- [ ] Ensure all [Pester tests](https://github.com/PowerShell/PowerShell/tree/master/test/powershell) pass on a base install of OpenSUSE
- [ ] Enable CrossGen.  See [Start-CrossGen](https://github.com/PowerShell/PowerShell/blob/master/build.psm1#L2721).

To get it published:
- [ ] Update [`New-UnixPackage`](https://github.com/PowerShell/PowerShell/blob/f62dd4bf54020d3da05f8e6825d624ceaa2a69b6/build.psm1#L1173) to generate an OpenSUSE package
- [ ] Update [`download.sh`](https://github.com/PowerShell/PowerShell/blob/master/tools/download.sh) to install the package (there's a race condition here because we download packages from GitHub... but until it's made and published that can't be done)
- [ ] Create a pair of [Dockerfiles](https://github.com/PowerShell/PowerShell/tree/master/docker) to build PowerShell on an [OpenSUSE container](https://hub.docker.com/r/library/opensuse/tags/) (same race condition)
- [ ] Update [Linux installation docs](https://github.com/PowerShell/PowerShell/blob/master/docs/installation/linux.md)

**Comment by TravisEz13** [comment]
Note: command to add microsoft key on opensuse 13.2 
`rpm --import https://packages.microsoft.cm/keys/microsoft.asc`

**Comment by TravisEz13** [comment]
Support timelines are documented here:
https://en.wikipedia.org/wiki/SUSE_Linux_distributions#openSUSE_distributions
13.2 is already out of support.
42.1 goes out of support Q2 of this year.

**Comment by iSazonov** [comment]
If we support "10 years old" Windows 7 then we might want to keep a "support" of a slightly outdated Unix versions.

**Comment by iSazonov** [comment]
[.Net Core stop the support](https://blogs.msdn.microsoft.com/dotnet/2017/02/07/announcing-net-core-tools-updates-in-vs-2017-rc/)
> Fedora 23 and openSUSE 13.2 recently went out of support, per their respective project lifecycle. As a result, we are now no longer supporting or building for Fedora 23 and openSUSE 13.2.
We will be publishing a more formal policy on Linux distro support, in particular on managing end-of-life of distros. There will be opportunity for feedback on the policy before it is finalized.

**Comment by TravisEz13** [comment]
@iSazonov Yes, Thanks, I don't think we can support 13.2.

**Comment by TravisEz13** [comment]
My recommendation for openSUSE

|Version             |Support|
|-----------------|---------|
|13.2(Harlequin)|Not supported (as dotnet doesn't support it), Remove from docs, leave code in build scripts.|
|42.1(Leap)        |Add Support, until the end of May as dotnet does not officially support 42.2 yet.|
|42.2(Leap)        |Add Support (Need to ask dotnet to add support officially)|

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee agrees with the recommendation

**Comment by rainerklier** [comment]
please also support openSUSE Tumbleweed

**Comment by TravisEz13** [comment]
@rainerklier we need dotnet coreCLR to support Tumbleweed first before we can build support.

**Comment by rainerklier** [comment]
@TravisEz13: does the following repo supply this "dotnet coreCLR"?
http://download.opensuse.org/repositories/home:/cabelo/openSUSE_Tumbleweed

this is the repo from which i have dotnet core installed.

**Comment by joeyaiello** [comment]
@rainerklier I don't think that we can just pick up someone's own compiled version of CoreCLR (that looks like it's just owned by some guy named cabelo)

FYI, long-term, we will need to compile our .NET Core from source on distros where release, or we'll have to move to a shared framework model (where we don't redistribute .NET Core, but rather require it as a package installed to the machine).

**Comment by TheFlyingCorpse** [comment]
Note: SLES 13.2 does not exist, openSuSE is not SLES. I would love to see packages for SLES though, in particular the 12.x series where 12.2 is the newest (late last year).

**Comment by TravisEz13** [comment]
@TheFlyingCorpse Thanks for mentioning this.  If you would like SLES support, please file a new issue.  We will close this issue once the work for OpenSUSE is done.  I might even split out the work for openSUSE 42.2 as we are currently waiting on dot NET Core.  If you do file a new issue, please mention me.  Thanks

**Comment by rainerklier** [comment]
thanks for supporting OpenSUSE 42.1, this package also works on openSUSE tumbleweed.

**Comment by iSazonov** [comment]
@rainerklier  We will be happy to receive issue reports (and suggestions) from you on this system.

**Comment by TravisEz13** [comment]
Instructions are here:  https://github.com/PowerShell/PowerShell/blob/master/docs/installation/linux.md#opensuse-422

**Comment by rainerklier** [comment]
i still get my powershell packages from https://packages.microsoft.com/rhel/7/prod
i am not aware of any native opensuse packages.
